linera-service/src/database_tool.rs

```diff
@@ -170,14 +184,35 @@ async fn evaluate_options(options: DatabaseToolOptions) -> Result<(), anyhow::Er
                 cache_size,
             };
             let full_storage_config = storage_config.add_common_config(common_config).await?;
-            let test = full_storage_config
-                .test_existence()
-                .await
-                .expect("successful delete_all operation");
+            let test = full_storage_config.test_existence().await?;
             if test {
-                process::exit(0);
+                tracing::info!("The database does exist");
+                return Ok(0);
             } else {
-                process::exit(1);
+                tracing::info!("The database does not exist");
+                return Ok(1);
```

Comment:
Can we return another value >=2 ?

Reply:
Yes, we can but then that would lead to the collision of the scenario of absence of table coinciding with the scenario of a failed connection. So, I do not think it is a good idea.